### PR TITLE
keep original filenames

### LINKED ISSUE #6: Add option to download assets with original filename
Currently stactask will download assets and name the file as the <asset_key>.<ext>, thereby keeping the extension, but possibly (probably) renaming the file.

Sometimes a subprocess will expect the file to be named the same as it was originally.   Present as option.

### PR DESCRIPTION
**Pull request by ircwaves**
Closes #6

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - ([#72]) Given that `_get_file` is part of the `AsyncFileSystem` spec, this
   adds the synchronous `get_file` as a way to retrieve files if `_get_file` is
   not found.
+- ([#74]) Added option `--keep-original-filenames` to support legacy
```
Comment (ircwaves) [inline CHANGELOG.md:15]:
```suggestion
- ([#77]) Added option `--keep-original-filenames` to support legacy
```

**Comment by ircwaves** [comment]
@matthewhanson @gadomski agree thatporting to `stac-asset` (#76)is definitely the better long-term move, but for immediate GTD,  I believe this addresses the issue in the interim.

File: pyproject.toml [hunk 2]
```diff
@@ -37,8 +37,8 @@ dev = [
     "pre-commit~=3.5.0",
     "pytest-cov~=4.1.0",
     "pytest~=7.4.0",
-    "ruff==0.1.9",
-    "types-setuptools~=68.2",
+    "ruff==0.1.11",
+    "types-setuptools~=69.0",
```
Comment (ircwaves) [inline pyproject.toml:41]: Looks like a rebase/merge fail here....that's already in main.

Comment (ircwaves) [reply pyproject.toml]: yep.  fixed.

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - ([#72]) Given that `_get_file` is part of the `AsyncFileSystem` spec, this
   adds the synchronous `get_file` as a way to retrieve files if `_get_file` is
   not found.
+- ([#77]) Added option `keep_original_filenames` to download routines to
```
Comment (gadomski) [inline CHANGELOG.md:15]:
```suggestion
- ([#77](https://github.com/stac-utils/stac-task/pull/77)) Added option `keep_original_filenames` to download routines to
```

**Review by gadomski — APPROVED** [review]

**Pull request merged**